pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -566,20 +579,38 @@ func (b *Build) BuildPod(images BuildPodImages, buildContext BuildContext) (*cor
 								return []string{fmt.Sprintf("-report=%s", ReportTOMLPath)}
 
 							}(),
+							func() []string {
+								if platformAPILessThan07 {
+									if b.Spec.RunImage.Image != "" {
+										return []string{"-run-image=" + b.Spec.RunImage.Image}
+									}
+								}
+								return []string{}
+							}(),
 							b.Spec.Tags),
 						VolumeMounts: volumeMounts([]corev1.VolumeMount{
 							layersMount,
 							workspaceVolume,
 							homeMount,
 							reportMount,
 						}, cacheVolumes),
-						Env: []corev1.EnvVar{
-							homeEnv,
-							{
-								Name:  platformAPIEnvVar,
-								Value: platformAPI.Original(),
+						Env: envs(
+							[]corev1.EnvVar{
+								homeEnv,
+								{
+									Name:  platformAPIEnvVar,
+									Value: platformAPI.Original(),
+								},
 							},
-						},
+							func() corev1.EnvVar {
+								if platformAPILessThan07 {
```

Comment:
Is there any harm in always having this env var?

Reply:
I don't think so.

Reply:
Fixed on [commit](https://github.com/pivotal/kpack/pull/988/commits/c190e3f3e8744b25ba10210409e1ce3ca58013c0)